kii_cloud_api.go

```diff
@@ -172,25 +229,110 @@ func (au *APIAuthor) OnboardGateway(request OnboardGatewayRequest) (*OnboardGate
 		return nil, err
 	}
 	req.Header.Set("content-type", "application/vnd.kii.onboardingWithVendorThingIDByThing+json")
-	req.Header.Set("authorization", "bearer " + au.Token)
+	req.Header.Set("authorization", "bearer "+au.Token)
 
-	client := &http.Client{}
-	resp, err := client.Do(req)
+	bodyStr, err := executeRequest(*req)
 	if err != nil {
 		return nil, err
+	} else {
+		err = json.Unmarshal(bodyStr, &ret)
+		if err != nil {
+			return nil, err
+		}
+		au.Token = ret.AccessToken
+		au.ID = ret.ThingID
+		return &ret, nil
 	}
-	defer resp.Body.Close()
+}
 
-	bodyStr, err := ioutil.ReadAll(resp.Body)
+// Request access token of end node of gateway.
+// When there's no error, GenerateEndNodeTokenResponse is returned.
+func (au APIAuthor) GenerateEndNodeToken(endnodeID string) (*GenerateEndNodeTokenResponse, error) {
```

Comment:
since this api doesn't change `APIAuthor` struct, so I use `au APIAuthor` for passing value not reference.